overlay.d/15fcos/usr/lib/systemd/system/coreos-fix-aleph-file.service

```
@@ -0,0 +1,13 @@
+# Remove after the next barrier release
+# https://github.com/coreos/fedora-coreos-tracker/issues/1724
+
+[Unit]
+Description=Remove extra attribute from aleph file
+
+[Service]
+Type=oneshot
+ExecStart=/usr/libexec/coreos-fix-aleph-file
+RemainAfterExit=yes
```

This needs to run in its own mount namespace to be able to remount `/sysroot` RW safely.

---

related thought : once we move to composeFS, we won't be able to do that kind of fixes anymore I guess

---

```suggestion
RemainAfterExit=yes
`RequireMountFor=/sysroot` 
```

Is that what you mean ? I don't think so but i don't find any hints in `man systemd.unit`

---

He means `MountFlags=slave`.

---

Yes, it's `MountFlags` sorry.

We will still be able to do that kind of fixes with composefs as this is `/sysroot`, the "real root" which is always writable by definition. Only `/`, the "composefs root" will be truly read only.

---

but then the signature won't match the underlying file ?

---

`/sysroot` is a mount of the root of the filesystem, not part of `/` (and so not covered by the composefs image/signature).